nxpo_hrms/public/js/leave_allocation.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Leave Allocation", {
+	refresh: function (frm) {
+		if (frm.doc.carry_forward) {
+			frm.toggle_display("custom_add_unused_leave", false);
+			frm.toggle_display("custom_total_unused_leave", false);
+		}else{
+			if (frm.doc.custom_is_add_unused_leave) {
+				// frm.set_df_property('custom_add_unused_leave', "read_only", 1);
+				frm.toggle_display("custom_add_unused_leave", false);
+				frm.toggle_display("custom_cancel_unused_leave", true);
+			}else{
+				frm.toggle_display("custom_add_unused_leave", true);
+				frm.toggle_display("custom_cancel_unused_leave", false);
+			}
+		}
+
+	},
+
+	custom_add_unused_leave: function(frm) {
+		if (!frm.doc.custom_is_add_unused_leave) {
+			frm.call({
+				doc: frm.doc,
+				method: 'get_leave_ledger',
```

Comment:
เนื่องจากไม่ต้อง extend class แล้ว
```suggestion
				method: 'nxpo_hrms.custom.leave_allocation.get_unused_leave',
```